e2e-tests/playwright/docs/accessibility/interaction_testing.md

```diff
@@ -0,0 +1,405 @@
+# Accessibility Interaction Testing Best Practices
+
+## Overview
+
+This guide covers **automated accessibility testing through user interaction simulation** using Playwright. These techniques test real user experiences with assistive technologies by automating keyboard navigation, focus management, and screen reader workflows.
+
+## Core Capabilities
+
+### **Keyboard Navigation Testing**
+
+- **Arrow Key Navigation** - Menu and list traversal
+- **Tab Navigation** - Focus order validation
+- **Enter/Space Activation** - Button and link interaction
+- **Escape Key Handling** - Modal and menu dismissal
+- **Letter Key Shortcuts** - Quick navigation patterns
+
+### **Focus Management Validation**
+
+- **Focus Visibility** - `:focus-visible` state verification
+- **Focus Trapping** - Modal and dialog containment
+- **Focus Restoration** - Return focus after interactions
+- **Focus Order** - Logical tab sequence validation
+
+### **State and Behavior Testing**
+
+- **ARIA State Changes** - `aria-expanded`, `aria-selected` verification
+- **Dynamic Content** - Live region announcements
+- **Error Handling** - Validation message accessibility
+- **Loading States** - Progress indication accessibility
+
+## Testing Patterns
+
+### **1. Keyboard Navigation Testing**
+
+```typescript
+test('keyboard navigation through menu', async ({page}) => {
+    // Open menu with keyboard
+    await page.locator('[data-testid="menu-button"]').focus();
+    await page.keyboard.press('Enter');
+
+    // Navigate through menu items
+    await expect(page.getByRole('menuitem').first()).toBeFocused();
+
+    await page.keyboard.press('ArrowDown');
+    await expect(page.getByRole('menuitem').nth(1)).toBeFocused();
+
+    await page.keyboard.press('ArrowDown');
+    await expect(page.getByRole('menuitem').nth(2)).toBeFocused();
+
+    // Test wrapping behavior
+    await page.keyboard.press('ArrowUp');
+    await expect(page.getByRole('menuitem').nth(1)).toBeFocused();
+
+    // Close menu with Escape
+    await page.keyboard.press('Escape');
+    await expect(page.getByRole('menuitem')).toHaveCount(0);
+
+    // Verify focus restoration
+    await expect(page.locator('[data-testid="menu-button"]')).toBeFocused();
+});
+```
+
+### **2. Focus Trapping in Modals**
```

Comment:
I don't think we need this example because this should be presumably written once for something like the GenericModal and then never needed again. 0/5 on if we actually remove it though since it still shows how to write a test